lib/main.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_arr = []
+    my_each do |x|
+      new_arr << x if yield(x) == true
+    end
+    new_arr
+  end
+
+  def my_all(arg = nil)
+    unless block_given?
+      falsy = 0
+      my_each do |x|
+        falsy += 1 if x == false || x.nil?
```

Comment:
have into consideration this test case too

` array.all?(Integer)`

and compare it with your method